src/cache/impl/LevelDbCache.ts

```diff
@@ -136,7 +136,37 @@ export class LevelDbCache<V = any> implements SortKeyCache<V> {
     return this.db as S;
   }
 
+  async getNumEntries(): Promise<Number> {
+    const keys = await this.db.keys().all();
+    return keys.length
+  }
+
   async prune(entriesStored = 5): Promise<PruneStats> {
-    throw new Error('Not implemented yet');
+    if (!entriesStored || entriesStored <= 0) {
+      entriesStored = 1
+    }
+
+    const keys = await this.db.keys().all();
```

Comment:
how about calling `this.allContracts` here? Instead of loading the contracts manually (which is exactly what `allContracts` is doing)